lib/random.gi

```diff
@@ -315,5 +315,6 @@ end)();
 # This method must rank below Random(SomeRandomSource, IsList)
 # for any random source SomeRandomSource, to avoid an infinite loop.
 InstallMethodWithRandomSource( Random, "for a random source and a (finite) collection",
-    [ IsRandomSource, IsCollection and IsFinite ], -8,
+    [ IsRandomSource, IsCollection and IsFinite ],
+    {} -> -RankFilter(IsCollection and IsFinite),
```

Comment:
IIRC, then me, @ChrisJefferson and others discussed this particular rank adjustment at length. The current static adjustment by -8 is not enough with all packages loaded. I think it's appropriate to rank this down to where the rank of the second argument is effectively zero. When GAP starts, this amounts to `-4` (so the rank actually of this method is actually a bit higher than it was before). Once all packages are loaded, it is `-126`... (but once the `automgrp` package is fixed to not have an implication `IsFinite => IsAmenable`, it'll be down to -6).

Alternatively, we could also use `{} -> -RankFilter(IsCollection and IsFinite)-RankFilter(IsRandomSource)`, where `RankFilter(IsRandomSource) = 2`, to effectively get the rank of this method down to 0, but that seems like overkill.